plugins/builds/update.js

```diff
@@ -51,8 +51,17 @@ module.exports = () => ({
 
                     // Users can only mark a running or queued build as aborted
                     if (!isBuild) {
+                        // Check if Screwdriver admin
+                        const adminDetails = request.server.plugins.banners
+                            .screwdriverAdminDetails(username, scmContext);
+
                         // Check desired status
-                        if (desiredStatus !== 'ABORTED') {
+                        if (adminDetails.isAdmin) {
+                            if (desiredStatus !== 'ABORTED' && desiredStatus !== 'FAILURE') {
+                                throw boom.badRequest('Admin can only update builds ' +
+                                    'to ABORTED or FAILURE');
+                            }
+                        } else if (desiredStatus !== 'ABORTED') {
                             throw boom.badRequest('Can only update builds to ABORTED');
```

Comment:
How about updating the msg to `User can only update builds to ABORTED` same like admin side msg?

Reply:
I fixed message.